src/mappings/swapsHandler.ts

```diff
@@ -1,43 +1,54 @@
-import {SubstrateExtrinsic} from "@subql/types";
-import {Swap} from "../types";
-import {formatU128ToBalance, checkIfExtrinsicExecuteSuccess, getExtrinsicNetworkFee, getExtrinsicId} from "./utils";
-import {LiquiditySourceType, SwapAmount} from "sora/api-interfaces";
+import {SubstrateExtrinsic} from '@subql/types';
+import {HistoryElement} from "../types";
 import type {Vec} from "@polkadot/types";
-
+import {checkIfExtrinsicExecuteSuccess, formatU128ToBalance, getExtrinsicNetworkFee, receiveSwapAmounts} from "./utils";
+import {LiquiditySourceType, SwapAmount} from "sora/api-interfaces";
 
 export async function handleSwaps(extrinsic: SubstrateExtrinsic): Promise<void> {
-    let record = new Swap(getExtrinsicId(extrinsic));
-    record.extrinsicHash = extrinsic.extrinsic.hash.toString();
-    record.blockHash = extrinsic.block.block.hash.toString();
-    record.address = extrinsic?.extrinsic.signer.toString();
-    record.networkFee = formatU128ToBalance(getExtrinsicNetworkFee(extrinsic));
-    record.selectedMarket = (extrinsic.extrinsic.args[4] as Vec<LiquiditySourceType>).map(lst => lst.toString());
-    record.datetime = extrinsic.block.timestamp;
-    let success = checkIfExtrinsicExecuteSuccess(extrinsic);
-    if (success) {
+    
+    logger.debug("Caught swap extrinsic")
+    
+    const record = new HistoryElement(extrinsic.extrinsic.hash.toString())
```

Comment:
this block of code is repeated in all handlers. Can it be moved to `utils`? 
I mean the lines from 11th to 19th.